Add req.method to passed tags

nsheaps opened the pull request:
Adds tracking for req.method to connect-datadog middleware that gets passed to statsd.histogram

rue-mchang committed:
new unit tests for connect-datadog library

rue-mchang committed:
make tests async

nsheaps committed:
Add tracking for request method type:

- Configurable with 'method' option on middleware initialization

rue-dhurd commented on lib/index.test.js:
```diff
@@ -201,6 +202,42 @@ describe('connectDatadog', () => {
+
+          // method option is explicitly set to false
+          await asyncConnectDatadogAndCallMiddleware({ response_code: true, method })
+          expectConnectedToDatadog(stat, statTags)
```
Not your fault, but I don't like how this method reads. Took me a minute to realize this was us asserting the method tag wasn't used. I think I may take that on in a followup PR 😉

rue-mchang commented on lib/index.js:
```diff
@@ -52,6 +53,10 @@ module.exports = function (options) {
+				statTags.push(`method:${req.method}`);
+			}
+
 			if (response_code) {
```
If you look up like 10 lines, this case is actually already being handled. We're just weren't _explicitly_ accepting `options.method` at the top. I like explicitly accepting `method` at the top, and I think we can just modify what's already there a few lines up to take `method` instead of `options.method`.

rue-dhurd replied:
If we don't have a use case for toggling it on and off, I don't think we should put in the option. This work was to add the method to the tag, so if it's already there 🎉 .

rue-mchang reviewed (changes requested):
There's some redundancy since `req.method` would get pushed to tags twice now.

nsheaps committed:
Remove duplicate `method` handling

It already existed so adding the code was not the appropriate path to take. In the mean time, I standardized the `method` option to be treated similarly to how the other options are treated

nsheaps committed:
Re-add mistakenly removed test

nsheaps committed:
Remove unnecessary setup on req.method

That’s now part of test setup for the request object

nsheaps committed:
Merge branch 'add-tests' into feature/add-req-method-tracking
